dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1552,12 +1585,15 @@ private boolean resolveHang(long tasksCompleted, boolean queueEmpty, int poolSiz
                     if (controllerCyclesWithoutHang > noHangCyclesThreshold) {
                         setPoolIncrementDecrement(poolSize);
                         hangBufferPoolSize -= poolDecrement;
+                        currentMinimumPoolSize = hangBufferPoolSize;
                         controllerCyclesWithoutHang = 0;
                     }
                 }
+            } else {
```

Comment:
I do not understand this else case.

Reply:
Based off of the logic, it seemed unnecessary to me which is why I brought it up.

Reply:
OK, deleted it.